Created new Thales KMIP topic for TDE

Andriciuc opened the pull request:
Added Thales integration, initial draft, to TDE.

Andriciuc committed:
Updated with an initial KMIP information

Added Thales integration, initial draft, to TDE.

dutow commented on contrib/pg_tde/documentation/docs/global-key-provider-configuration/kmip-thales.md:
```diff
@@ -0,0 +1,28 @@
+# Thales KMIP Server Configuration
+
+To use the Python library pykmip for cryptographic operations with Thales CipherTrust Manager, see Using pykmip in the Implementing Thales CipherTrust Manager documentation for instructions. pykmip is a Python library that implements the KMIP industry standard for key management operations.
```
pykmip has nothing to do with thales

Andriciuc replied:
Thank you! Was a dummy statement, replaced with proper intro: '

`pg_tde` is compatible with the Thales CipherTrust Manager via the KMIP protocol. For a full setup guide, see [the following documentation] (tbd).

dutow commented on contrib/pg_tde/documentation/docs/global-key-provider-configuration/kmip-thales.md:
````diff
@@ -0,0 +1,28 @@
+```sql
+SELECT pg_tde_add_global_key_provider_kmip(
+    'thales_kmip_provider',
+    'kmip.thales.local',
````
it's better to use kmip.example.org or something like that here, that makes it clearer that this is an examle domain

Andriciuc replied:
added kmip.example.org

dutow commented on contrib/pg_tde/documentation/docs/global-key-provider-configuration/kmip-thales.md:
````diff
@@ -0,0 +1,28 @@
+```
+
+!!! note
+    Replace the above paths with the actual certificate locations on your PostgreSQL host.
````
not just the paths, the other parameters too

Andriciuc replied:
Reworded:

    Replace the example paths and parameters with the actual certificate locations on your PostgreSQL host.

Andriciuc committed:
Merge branch 'TDE_REL_17_STABLE' into docs-add-thales

Andriciuc committed:
updated thales with more information and feedback

Andriciuc committed:
Merge branch 'TDE_REL_17_STABLE' into docs-add-thales

Andriciuc commented on contrib/pg_tde/documentation/docs/global-key-provider-configuration/kmip-thales.md:
```diff
@@ -0,0 +1,26 @@
+
+`pg_tde` is compatible with the Thales CipherTrust Manager via the KMIP protocol. For a full setup guide, see [the following documentation](https://thalesdocs.com/ctp/cm/2.19/reference/kmip-ref/index.html?).
+
+## Recommended Configuration Steps
```
I am unsure if this is enough information to give to the user, if we can have proper steps to setup Thales with pg_tde, I would suggest we use them here. The documentation available online is lacking clarity heavily on the config part

Andriciuc replied:
For now I will leave it as is, more information will  be added as we move forwards

Andriciuc committed:
Merge branch 'TDE_REL_17_STABLE' into docs-add-thales

Andriciuc committed:
Merge branch 'TDE_REL_17_STABLE' into docs-add-thales

dutow commented on contrib/pg_tde/documentation/docs/global-key-provider-configuration/kmip-thales.md:
```diff
@@ -0,0 +1,26 @@
+## Recommended Configuration Steps
+
+1. Obtain and secure the certificates from Thales CipherTrust Manager.
+2. Create `pykmip.conf`.
```
No need for pykmip conf, that's for pykmip

Andriciuc replied:
Thank you @dutow ! I decided to keep it basic and follow Fortanix example, removed config steps and set a link for adding/configuring the example config

dutow commented on contrib/pg_tde/documentation/docs/global-key-provider-configuration/kmip-thales.md:
```diff
@@ -0,0 +1,26 @@
+1. Obtain and secure the certificates from Thales CipherTrust Manager.
+2. Create `pykmip.conf`.
+3. Configure PostgreSQL for pg_tde + KMIP.
+4. Create or Retrieve the KMIP Key.
```
No need to do that, that's handled by postgres + pg_tde

Andriciuc replied:
Removed the steps.

Andriciuc committed:
Merge branch 'TDE_REL_17_STABLE' into docs-add-thales

Andriciuc committed:
Update kmip-thales.md

removed the configuration steps and example config and set a link to our already existing docs for the user to see kmip config example

Andriciuc committed:
Merge branch 'docs-add-thales' of https://github.com/percona/postgres into docs-add-thales

Andriciuc committed:
Merge branch 'TDE_REL_17_STABLE' into docs-add-thales

dutow commented on contrib/pg_tde/documentation/docs/global-key-provider-configuration/kmip-thales.md:
```diff
@@ -0,0 +1,8 @@
+# Thales KMIP Server Configuration
+
+`pg_tde` is compatible with the Thales CipherTrust Manager via the KMIP protocol. For a full setup guide, see [the following documentation](https://thalesdocs.com/ctp/cm/2.19/reference/kmip-ref/index.html?).
```
maybe the `thales documentation` instead of following, to make clear that it's not our documentation?

Andriciuc replied:
It can be done as that too, yes, replaced and also removed the below note as we don't use config example

Andriciuc committed:
Update kmip-thales.md

removed note as we removed the config ex, and replaced following with Thales docs

Andriciuc committed:
Merge branch 'TDE_REL_17_STABLE' into docs-add-thales

Andriciuc committed:
Merge branch 'TDE_REL_17_STABLE' into docs-add-thales

Andriciuc committed:
Merge branch 'TDE_REL_17_STABLE' into docs-add-thales
